src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -419,15 +447,21 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                             var matchingNode = nodes.Where(x => x.GUID == inputGuid).FirstOrDefault();
                             if (matchingNode != null)
                             {
-                                matchingNode.IsSetAsInput = true;
+                                matchingNode.IsSetAsInput = updateIsSetAsInput;
+                                matchingNode.IsSetAsOutput = updateIsSetAsOutput;
+                                matchingNode.IsFrozen = updateIsFrozen;
+                                matchingNode.UpdateValue(new UpdateValueParams("IsVisible", updateIsVisible.ToString()));
```

Comment:
I understand the Frozen aspect of CLI usage but not about `IsVisible`, can you expand a bit here?

Reply:
I think we need to have a discussion around this... I feel this muddies our API separation and while useful could also be solved by CLI creating a viewModel.